### PR TITLE
First commits

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This initial pull request adds a workflow to represent the current `lint` job from [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic) (and subsequently by all of our repositories that use the [Lineage](https://github.com/cisagov/action-lineage) system).

## 💭 Motivation and context ##

Migrating our GitHub Actions tasks that are common across repositories to [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) is a maintenance and updating win. Once our other repositories are configured to use the `common-lint.yml` workflow any updates to the workflow here will be available without substantial modification to our other projects using this workflow.

## 🧪 Testing ##

Automated tests pass. I created a [branch in cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic/tree/testing/lint_job_as_reusable_workflow) to test this and the [Action run](https://github.com/cisagov/skeleton-generic/actions/runs/3513124963) ran successfully using the workflow added in this pull request.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [ ] Add a tag or create a release.

### CODE REVIEW

**Review by jsf9k — CHANGES_REQUESTED** [review]
This looks great to me.  It's a great improvement and a step down the right path.  I'll be glad to see some of the GH Actions stuff DRY out a bit.

I have one (very minor) requested change:
- Document in the README why we made the decision to store multiple workflows in a single repo.

I have two questions for @team-ois's consideration:
- Is it worth documenting in the README the other workflows that the individual workflows will pull down and run?
- Even though they are small jobs, there may be some benefit to breaking out the `Install Packer`, `Install shfmt`, and `Install terraform-docs` into their own workflows (likely in separate repos) so that other non-CISA folks can easily make use of them.  That doesn't have to happen now, but if the @team-ois consensus is that it's a good idea then I can create issues.

**Comment by mcdonnnj** [comment]
> This looks great to me. It's a great improvement and a step down the right path. I'll be glad to see some of the GH Actions stuff DRY out a bit.
> 
> I have one (very minor) requested change:
> 
>     * Document in the README why we made the decision to store multiple workflows in a single repo.
> 
> 
> I have two questions for @team-ois's consideration:
> 
>     * Is it worth documenting in the README the other workflows that the individual workflows will pull down and run?
> 
>     * Even though they are small jobs, there may be some benefit to breaking out the `Install Packer`, `Install shfmt`, and `Install terraform-docs` into their own workflows (likely in separate repos) so that other non-CISA folks can easily make use of them.  That doesn't have to happen now, but if the @team-ois consensus is that it's a good idea then I can create issues.

Regarding `Install Packer` I would much rather we finish what I started in [cisagov/setup-packer](https://github.com/cisagov/setup-packer) (forked [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) and started changing it to install Packer instead). I would also say that the `Install shfmt` and `Install terraform-docs` steps could be a more generic "Install Go tool" composite Action that is then used in workflows as needed.

**Comment by jsf9k** [comment]
> Regarding `Install Packer` I would much rather we finish what I started in [cisagov/setup-packer](https://github.com/cisagov/setup-packer) (forked [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) and started changing it to install Packer instead). I would also say that the `Install shfmt` and `Install terraform-docs` steps could be a more generic "Install Go tool" composite Action that is then used in workflows as needed.

I'm fine with both those ideas.

File: .github/workflows/common-lint.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            py${{ steps.setup-python.outputs.python-version }}-\
+            go${{ steps.setup-go.outputs.go-version }}-\
+            packer${{ needs.get_tool_versions.outputs.packer-version }}-\
+            tf${{ needs.get_tool_versions.outputs.terraform-version }}-"
+        with:
+          # Note that the .terraform directory IS NOT included in the
+          # cache because if we were caching, then we would need to use
+          # the `-upgrade=true` option. This option blindly pulls down the
+          # latest modules and providers instead of checking to see if an
+          # update is required. That behavior defeats the benefits of caching.
```
Comment (dav3r) [inline .github/workflows/common-lint.yml]: Punctuation fix:
```suggestion
          # update is required. That behavior defeats the benefits of caching,
```

**Review by dav3r — APPROVED** [review]
This is excellent work!  I agreement with @jsf9k's [request to document our decision to store multiple workflows in this repo](https://github.com/cisagov/github-actions-workflows/pull/14#pullrequestreview-1188503215) and with @mcdonnnj's [ideas for installing packer and go-based tools](https://github.com/cisagov/github-actions-workflows/pull/14#issuecomment-1322289879).

**Comment by mcdonnnj** [comment]
This branch has been changed far beyond the original purpose of this PR. I am closing this PR so that an appropriately described new PR can be created.

**Pull request closed without merging**